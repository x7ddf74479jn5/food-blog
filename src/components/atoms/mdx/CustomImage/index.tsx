import Image from "next/future/image";
import type { ComponentPropsWithoutRef, FC } from "react";

type CustomImageProps = ComponentPropsWithoutRef<"img">;

export const CustomImage: FC<CustomImageProps> = ({ src = "", alt = "", width, height }) => {
  if (width && height) {
    return (
      <div className="relative mb-4 flex w-full items-center justify-center">
        <Image
          src={src}
          alt={alt}
          width={width}
          height={height}
          // placeholder="blur"
          // blurDataURL={""}
          className="h-auto max-w-full"
        />
      </div>
    );
  }

  return (
    <div className="relative mb-4 flex h-48 w-full items-center justify-center lg:h-96">
      <Image src={src} alt={alt} sizes="100vw" fill className="object-contain" placeholder="blur" />
    </div>
  );
};
