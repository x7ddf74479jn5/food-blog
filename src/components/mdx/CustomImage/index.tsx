import type { ImageProps } from "next/image";
import Image from "next/image";

type Props = {
  className: string;
  alt: string;
} & ImageProps;

const CustomImage: React.VFC<Props> = (props) => {
  return (
    <div className="next-image-container">
      <Image
        className={props.className}
        src={props.src}
        alt={props.alt}
        width={props.width}
        height={props.height}
        loading={props.loading || "lazy"}
        layout="responsive"
      />
    </div>
  );
};

export default CustomImage;
