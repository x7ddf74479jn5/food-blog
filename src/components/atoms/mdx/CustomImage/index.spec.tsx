import renderer from "react-test-renderer";

import CustomImage from ".";

describe("components/atoms/mdx/CustomImage", () => {
  it("snapshot", () => {
    const image = renderer.create(<CustomImage src={"/mocks/data/images/2996666_s.jpg"} />).toJSON();
    expect(image).toMatchSnapshot();
    const nextImage = renderer
      .create(<CustomImage src={"/mocks/data/images/2996666_s.jpg"} width={100} height={100} />)
      .toJSON();
    expect(nextImage).toMatchSnapshot();
  });
});
