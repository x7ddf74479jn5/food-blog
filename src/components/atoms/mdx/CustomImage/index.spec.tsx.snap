// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/mdx/CustomImage snapshot 1`] = `
<div
  className="flex relative justify-center items-center mb-4 w-full h-48 lg:h-96"
>
  <div
    style={
      Object {
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "left": 0,
        "margin": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <img
      alt=""
      data-nimg="fill"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "contain",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=3840&q=75"
        srcSet="/_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=640&q=75 640w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=750&q=75 750w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=828&q=75 828w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=3840&q=75 3840w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </div>
</div>
`;

exports[`components/atoms/mdx/CustomImage snapshot 2`] = `
<div
  className="flex relative justify-center items-center mb-4 w-full"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "margin": 0,
        "maxWidth": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "maxWidth": "100%",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
        style={
          Object {
            "border": "none",
            "display": "block",
            "margin": 0,
            "maxWidth": "100%",
            "padding": 0,
          }
        }
      />
    </div>
    <img
      alt=""
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt=""
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=128&q=75 1x, /_next/image?url=%2Fmocks%2Fdata%2Fimages%2F2996666_s.jpg&w=256&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </div>
</div>
`;
